### ISSUE TITLE
When using CommonMarkParser code blocks do not get code highlighting

### ISSUE THREAD

**Issue by jerel** [rtfd/readthedocs.org#1485]
Example:

    ``` python
    def function():
        pass
    ```
renders as `<pre class="literal-block"></pre>` with no code styling classes applied. 

![highlighting](https://cloud.githubusercontent.com/assets/322706/8888409/d891e61a-3271-11e5-90fb-87c8d479e38a.png)

**Comment by gregmuellegger** [comment]
`CommonMark` seems to do the right thing:

```
>>> import CommonMark
>>> parser = CommonMark.DocParser()
>>> renderer = CommonMark.HTMLRenderer()
>>> renderer.render(parser.parse("""
... ``` python
... def function():
...     pass
... ```
... """))
'<pre><code class="language-python">def function():\n    pass\n</code></pre>\n'
```

But you are referencing python code in the ticket description and show JSON output in the screenshot. Could you give us a link to the page where the rendering breaks for you?

**Comment by jerel** [comment]
Sure, here is a link to the docs page http://django-rest-framework-json-api.readthedocs.org/en/latest/usage.html#configuration I suppose CommonMark is doing what it is supposed to do but its output differs from the `rst` code blocks so there's no highlighting style in the readthedocs.org theme.

Also note that the `Edit on Github` button on the page I linked to has `.rst` hardcoded so returns a 404 for `.md`. Here's the correct link https://github.com/django-json-api/django-rest-framework-json-api/blob/develop/docs/usage.md

**Comment by ericholscher** [comment]
Support for this was just merged into our upstream, and should be pushed to
RTD with a new release of recommonmark:
https://github.com/ericholscher/recommonmark/pull/1

On Mon, Jul 27, 2015 at 8:41 AM, Jerel Unruh <notifications@github.com>
wrote:

> Sure, here is a link to the docs page
> http://django-rest-framework-json-api.readthedocs.org/en/latest/usage.html#configuration
> I suppose CommonMark is doing what it is supposed to do but its output
> differs from the rst code blocks so there's no highlighting style in the
> readthedocs.org theme.
>
> Also note that the Edit on Github button on the page I linked to has .rst
> hardcoded so returns a 404 for .md. Here's the correct link
> https://github.com/django-json-api/django-rest-framework-json-api/blob/develop/docs/usage.md
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rtfd/readthedocs.org/issues/1485#issuecomment-125249091>
> .
>



-- 
Eric Holscher
Maker of the internet residing in Portland, Oregon
http://ericholscher.com

**Comment by gregmuellegger** [comment]
That commit fixed it https://github.com/rtfd/recommonmark/commit/4005ba5f2f19112fa1469b748d48a3a4ec01822f
@ericholscher when will a new release happen?

**Comment by ericholscher** [comment]
Just released 0.2.0 of recommonmark and updated our dep. It will be resolved with next deploy.

**Comment by jerel** [comment]
Thanks so much guys. It looks very nice now with markdown
